src/borg/repository.py

```diff
@@ -872,7 +879,7 @@ def report_error(msg):
                 continue
             try:
                 objects = list(self.io.iter_objects(segment))
-            except IntegrityError as err:
+            except (IntegrityError, LoggedIO.AtticSegment) as err:
                 report_error(str(err))
                 objects = []
                 if repair:
```

Comment:
I just realized that we will actually never catch `AtticSegment` here. We already call `get_segments_transaction_id()` much earlier in the `check()` function. So if we decide to catch that error during `check`, it needs to be done much earlier. I tend to remove the extra check for `AtticSegment` here (leave this part of code untouched) and add proper Attic repository handling to `check` instead.